packages/amplify_authenticator/example/integration_test/sign_up_with_username_test.dart

```diff
@@ -45,106 +45,70 @@ void main() {
 
     // Scenario: Login mechanism set to "username"
     testWidgets('Login mechanism set to "username"', (tester) async {
-      await tester.loadAuthenticator(authenticator);
-      await tester.navigateToSignUp();
-
-      // Then I see "Username" as an input field
-      expect(
-        // Match on the title of the username field since it is dependent on
-        // the username alias selected.
-        find.descendant(
-          of: find.byKey(keyUsernameSignUpFormField),
-          matching: find.text('Username'),
-        ),
-        findsOneWidget,
-      );
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectUserNameIsPresent();
     });
 
     // Scenario: "Preferred Username" is included from `aws_cognito_signup_attributes`
     testWidgets(
         '"Preferred Username" is included from `aws_cognito_signup_attributes`',
         (tester) async {
-      await tester.loadAuthenticator(authenticator);
-      await tester.navigateToSignUp();
-
-      // Then I see "Username" as an input field
-      expect(find.byKey(keyPreferredUsernameSignUpFormField), findsOneWidget);
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectPreferredUserNameIsPresent();
```

Comment:
This step of the test seems to be flaky. This is the case in the amplify-authenticator branch as well. I am having trouble figuring out why, but since it is unrelated I am going to fix it in a follow up PR.

Reply:
Addressed with https://github.com/aws-amplify/amplify-flutter/pull/1125/commits/e5b8ffffe16c69fc2b6d222c547a6a32c3957fca